launchcontainer/launchcontainer.py

```diff
@@ -104,65 +105,80 @@ class LaunchContainerXBlock(XBlock):
-                # A string: the currently supported implementation.
-                (
-                    "ENV_TOKENS[{}]".format(WHARF_URL_KEY),
-                    settings.ENV_TOKENS.get(WHARF_URL_KEY)
-                ),
-                # A dict: the deprecated version.
-                (
-                    "ENV_TOKENS['LAUNCHCONTAINER_API_CONF']",
-                    settings.ENV_TOKENS.get('LAUNCHCONTAINER_API_CONF', {}).get('default')
-                ),
-                # Fallback to the default.
-                (
-                    "Default", DEFAULT_WHARF_URL
-                )
+            site = Site.objects.get(domain=edx_site_domain)
+        except Site.DoesNotExist:
+            site = get_current_site()  # From the request.
+        else:
+            site = site.domain
+
+        url = cache.get(make_cache_key(site))
+
+        if url:
+            return url
+        elif not url and site.configuration:
+            site_wharf_url = site.configuration.get_value(WHARF_URL_KEY)
+        elif not url and siteconfig_helpers:
+            site_wharf_url = siteconfig_helpers.get_value(WHARF_URL_KEY)
+
+        # Nothing in the cache. Go find the URL.
+        urls = (
+            # A SiteConfig object: this is the preferred implementation.
+            (
+                'SiteConfiguration', site_wharf_url
+            ),
+            # TODO: Maybe we can cache this and set up a signal
+            # to update this value if a SiteConfig object is changed.
+            # A string: the currently supported implementation.
+            (
+                "ENV_TOKENS[{}]".format(WHARF_URL_KEY),
+                settings.ENV_TOKENS.get(WHARF_URL_KEY)
+            ),
+            # A dict: the deprecated version.
+            (
+                "ENV_TOKENS['LAUNCHCONTAINER_API_CONF']",
+                settings.ENV_TOKENS.get('LAUNCHCONTAINER_API_CONF', {}).get('default')
+            ),
+            # Fallback to the default.
+            (
+                "Default", DEFAULT_WHARF_URL
             )
+        )
 
-            validator = validators.URLValidator()
+        validator = validators.URLValidator()
 
-            def is_valid(url):
-                try:
-                    validator(url)
-                except validators.ValidationError:
-                    return False
-                else:
-                    return True
-
-            logger.debug("XBlock-launchcontainer urls attempted: {}".format(urls))
-            if IS_OPENEDX_ENVIRON:
-                logger.debug("Current site: {}".format(get_current_site()))
-                logger.debug("Current site config enabled: {}".format(
-                    is_site_configuration_enabled())
-                )
-                logger.debug("Current site config LAUNCHCONTAINER_WHARF_URL: {}".format(
-                    siteconfig_helpers.get_value('LAUNCHCONTAINER_WHARF_URL')))
+        def is_valid(url):
```

Comment:
Maybe move this out of here to the `utils.py` file or somewhere similar, so the actual code is clearer.